### PR TITLE
Charge solving, imaging and configuration 

### PR DESCRIPTION
**Pull request by brettviren**
This PR is meant to be a checkpoint in the work on charge solving, imaging and new "layered" configuration pattern.

Summary of what this PR adds:

- `.zip` is now a supported stream format from custard so we can make compressed `.npz` files
- Boost multi-array is now supported for stream I/O.
- Initial support for saving `ICluster` as set of Numpy arrays.  
- Update to JSON format for `ICluster` to accommodate presence of new "activity uncertainty".
- Removal of old `JsonClusterTap`, the `ClusterFileSink` contains the functionality plus can save to custard streams.
- All the `Simple*` data classes are finally moved out of `iface` and in to `aux`.
- Remove obsolete, wrongly placed configuration.
- Remove false configuration "helpers" which were somewhat a false start.
- Began `cfg/layers/` which is the true start of a new configuration pattern which supports creation of detector-independent high-layer Jsonnet config and is generally better organized.
- Use new `MaskSlice` in the new `ChargeSolving`

See related: https://github.com/WireCell/wire-cell-python/pull/10

### CODE REVIEW

**Comment by brettviren** [comment]
This is now up to date with `master` as of release `0.20.0`.

File: sio/src/ClusterFileSink.cxx [hunk 1]
```diff
@@ -39,26 +38,84 @@ WireCell::Configuration Sio::ClusterFileSink::default_configuration() const
...
+    return ss.str();
+}
+
+void Sio::ClusterFileSink::jsonify(const ICluster& cluster)
+{
+    auto top = Aux::jsonify(cluster);
+    std::stringstream topss;
+    topss << top;
+    auto tops = topss.str();
+    m_out << "name " << fqn(cluster, "graph", "json") << "\n"
```
Comment (HaiwangYu) [inline sio/src/ClusterFileSink.cxx:60]: is this needed or just for debugging?

Comment (HaiwangYu) [reply sio/src/ClusterFileSink.cxx:60]: This was explained by @brettviren

**Comment by HaiwangYu** [comment]
Simulation and SigProc look consistent with #165 

Screen Shot 2022-06-22 at 12 15 30 AM
Screen Shot 2022-06-22 at 12 15 39 AM

**Pull request merged**